### PR TITLE
Add `isStorePath` tests

### PR DESCRIPTION
**Pull request by Profpatsch**
Some tests cc @zimbatm

### CODE REVIEW

File: lib/tests/misc.nix [hunk 1]
```diff
@@ -88,6 +87,35 @@ runTests {
...
   };
 
+  testIsStorePath =  {
+    expr =
+      let goodPath =
+            "/nix/store/d945ibfx9x185xf04b890y4f9g3cbb63-python-2.7.11";
+      in {
+        storePath = isStorePath goodPath;
+        storePathAppendix = isStorePath
+          "${goodPath}/bin/python";
```
Comment (zimbatm) [inline lib/tests/misc.nix]: why isn't that a store path?

File: lib/tests/misc.nix [hunk 2]
```diff
@@ -88,6 +87,35 @@ runTests {
...
+        nonAbsolute = isStorePath (concatStrings (tail (stringToCharacters goodPath)));
```
Comment (zimbatm) [inline lib/tests/misc.nix]: maybe add `otherPath = isStorePath "/something/else"`?

**Review by zimbatm — APPROVED** [review]
Didn't I already merge this function?

**Comment by Profpatsch** [comment]
I wonder why it includes the first commit again, yes. Maybe the commit ID changed?

File: lib/tests/misc.nix (hunk 1 above)
Comment (Profpatsch) [inline lib/tests/misc.nix]: I wonder. We could change the semantics of the function. @edolstra would that make sense?
In any case, it’s nice that this test case documents the current semantics.

File: lib/tests/misc.nix (hunk 2 above)
Comment (Profpatsch) [inline lib/tests/misc.nix]: Done.

File: lib/tests/misc.nix (hunk 1 above)
Comment (zimbatm) [inline lib/tests/misc.nix]: Agreed, without the tests we wouldn't be having this conversation.

**Pull request merged**